### PR TITLE
[Refactor]  Remove OCMock from ZMOperationLoop tests

### PR DESCRIPTION
**Pull request by typfel**
## What's new in this PR?

Remove OCMock from ZMOperationLoop tests by

- Removing ZMSyncStategy as direct dependency
   * [x] Move MOC save operation to SyncStrategy 
   * [x] Update sync status directly when push channel open/closes
   * [x] Don't expose syncStrategy as a public property on the operation loop
- Rewrite tests to use plain old swift mocks
   * [x] ZMSyncStrategy
   * [x] TransportSession
   * [x] ApplicationStatusDirectory

### CODE REVIEW

File: Tests/Source/UserSession/ZMUserSessionTestsBase.m [hunk 1]
```diff
@@ -97,14 +97,15 @@ - (void)setUp
 
     self.operationLoop = [OCMockObject mockForClass:ZMOperationLoop.class];
     [[self.operationLoop stub] tearDown];
-    [[[self.operationLoop stub] andReturn:self.syncStrategy] syncStrategy];
+//    [[[self.operationLoop stub] andReturn:self.syncStrategy] syncStrategy];
```
Comment (KaterinaWire) [inline Tests/Source/UserSession/ZMUserSessionTestsBase.m:100]: I believe we can remove this comment

Comment (David-Henner) [inline Tests/Source/UserSession/ZMUserSessionTestsBase.m:100]: is this comment wanted?

**Review by David-Henner — APPROVED** [review]

File: Tests/Source/UserSession/ZMUserSessionTestsBase.m (hunk 1 above)
Comment (typfel) [reply Tests/Source/UserSession/ZMUserSessionTestsBase.m]: 👍

**Review by KaterinaWire — APPROVED** [review]

**Comment by typfel** [comment]
@zenkins release patch

**Pull request merged**